docs/releases/upgrading/upgrade-v4.0.md

```diff
@@ -0,0 +1,41 @@
+# Upgrading from v3.0 to v4.0
+
+
+## Prerequisites
+
+The steps listed in this article require an existing local installation of InvenioRDM 3.0, please make sure that this is given!
+If unsure, run `invenio-cli install` from inside the instance directory before executing the listed steps.
+
+**Note**: Do *not* delete the old Python virtual environment, or the database migration may complain about missing packages.
+
+!!! warning "Backup"
+
+    Always backup your database and files before you try to perform an upgrade.
+
+!!! warning "Upgrade from v1 to v2"
+
+    If you are upgrading an instance, that you previously upgrade from v1, please
+    check the [troubleshooting section](/releases/upgrading/upgrade-v2.0/#troubleshooting) for
+    errors that were discovered in the v1 to v2 upgrade.
+
+## Upgrade Steps
+
+First, the Elasticsearch indices are deleted and the packages are upgraded. Then, the upgrade command will be executed, this will migrate the database, run the custom migration script and rebuild the Elasticsearch indices .
+
+This can be achieved by the following Bash shell commands:
+
+~~~bash
+# NOTE: make sure you're in the instance directory
+
+# Delete ES indices
+pipenv run invenio index destroy --yes-i-know
```

Comment:
Is this needed before?

Woudl be best if we could avoid it or at least make it before upgrade and after the package update

Reply:
It's needed to delete the indexes `rdmrecords-drafts-draft-v1.0.0` and  `rdmrecords-records-record-v1.0.0`. I guess we can move this to the migration and we can remove it from the beginning. I will have a look tomorrow.